src/axom/bump/utilities/utilities.hpp

```diff
@@ -0,0 +1,197 @@
+ *        determine the type that should be used to accumulate values when we
+ *        do floating point math.
+ *
+ * \note this belongs in algorithm utilities, maybe core.
+ */
+template <typename T>
+struct accumulation_traits
+{
+  using type = float;
+};
+
+template <>
+struct accumulation_traits<double>
+{
+  using type = double;
+};
+
+template <>
+struct accumulation_traits<long>
+{
+  using type = double;
+};
+
+template <>
+struct accumulation_traits<unsigned long>
+{
+  using type = double;
+};
+
+//------------------------------------------------------------------------------
+/*!
+ * \brief Fill an ArrayView with a value.
+ *
+ * \tparam ExecSpace The execution space where the fill will be done.
+ * \tparam T The data type of the values in the ArrayView.
+ *
+ * \param view The ArrayView being filled.
+ * \param fillValue The value to be used for filling the ArrayView.
+ */
+template <typename ExecSpace, typename T>
+void fill(axom::ArrayView<T> view, T fillValue)
+{
+  axom::for_all<ExecSpace>(
+    view.size(),
+    AXOM_LAMBDA(axom::IndexType index) { view[index] = fillValue; });
+}
+
+//------------------------------------------------------------------------------
+/*!
+ * \brief Use binary search to find the index of the \a value in the supplied
+ *        sorted view.
+ *
+ * \tparam T The type of elements we're handling.
+ * \tparam ContainerT A container or view of T.
+ *
+ * \param[in] value The search value.
+ * \param[in] view A view or container that contains the sorted search data values.
+ *
+ * \return The index where value was located in view or -1 if not found.
+ */
+template <typename T, typename ContainerT>
+AXOM_HOST_DEVICE std::int32_t bsearch(T value, const ContainerT &view)
+{
+  std::int32_t index = -1;
+  std::int32_t left = 0;
+  std::int32_t right = view.size() - 1;
+  while(left <= right)
+  {
+    std::int32_t m = (left + right) / 2;
+    if(view[m] < value)
+      left = m + 1;
+    else if(view[m] > value)
+      right = m - 1;
+    else
+    {
+      index = m;
+      break;
+    }
+  }
+
+  return index;
+}
+
+//------------------------------------------------------------------------------
+/*!
+ * \brief Hash a stream of bytes into a uint64_t hash value.
+ *
+ * \param[in] data The bytes to be hashed.
+ * \param[in] length The number of bytes to be hashed.
+ *
+ * \return A uint64_t hash for the byte stream.
+ *
+ * \note The byte stream is hashed using a Jenkins-hash algorithm forwards and
+ *       backwards and the two results are merged into a uint64_t. The length is
+ *       also part of the hash to guard against a lot of repeated values in the
+ *       byte stream hashing to the same thing.
+ *
+ * \note We make this function inline since it is not a template and we want to
+ *       use it in both host and device code.
+ */
+AXOM_HOST_DEVICE
+inline std::uint64_t hash_bytes(const std::uint8_t *data, std::uint32_t length)
```

Comment:
Should this move to `core`?

Does it relate to https://github.com/LLNL/axom/pull/1583 ?